lib/dialects/mssql/query-generator.js

```diff
@@ -680,6 +680,25 @@ var QueryGenerator = {
     return sql;
   },
 
+  getPrimaryKeyConstraint(table, attributeName) {
+    const tableName = wrapSingleQuote(table.tableName || table);
+    const sql = [
```

Comment:
I have no issue with Array based query generation and joining it, I would mostly encourage it for query generation.

My point was you put join result in a variable and return that variable, why don't directly return output of join directly